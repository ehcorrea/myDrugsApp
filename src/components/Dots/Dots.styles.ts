import styled, { css } from 'styled-components/native';

export const Container = styled.View`
  flex-direction: row;
`;

export const Bar = styled.View`
  ${({ theme }) => css`
    background-color: ${theme.colors.black};
    border-radius: ${theme.border.radius.medium}px;
    height: ${theme.spacings.xxsmall}px;
    width: ${theme.spacings.large}px;
  `}
`;

export const Dot = styled(Bar)`
  ${({ theme }) => css`
    background-color: ${theme.colors.gray};
    height: ${theme.spacings.xxsmall}px;
    margin-left: ${theme.spacings.xxsmall / 4}px;
    width: ${theme.spacings.xxsmall}px;
  `}
`;
