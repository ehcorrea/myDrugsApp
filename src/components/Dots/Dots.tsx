import React from 'react';

import * as S from './Dots.styles';

export type DotsProps = {
  dots?: number;
};

export function Dots({ dots = 3, ...props }: DotsProps) {
  return (
    <S.Container {...props}>
      <S.Bar />
      {Array(dots)
        .fill(0)
        .map((_, index) => (
          <S.Dot key={_ + index} />
        ))}
    </S.Container>
  );
}
