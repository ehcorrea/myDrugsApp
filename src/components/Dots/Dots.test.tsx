import { render } from '@/test/utils';
import { Dots } from './Dots';

test.each([undefined, 1, 2, 3])('render dots with %s', (value) => {
  const container = render(<Dots dots={value} />);

  expect(container).toMatchSnapshot();
});
