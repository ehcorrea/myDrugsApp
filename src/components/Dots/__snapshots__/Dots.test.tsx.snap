// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render dots with 1 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#030517",
          "borderRadius": 10,
          "height": 8,
          "width": 20,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
</View>
`;

exports[`render dots with 2 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#030517",
          "borderRadius": 10,
          "height": 8,
          "width": 20,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
</View>
`;

exports[`render dots with 3 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#030517",
          "borderRadius": 10,
          "height": 8,
          "width": 20,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
</View>
`;

exports[`render dots with undefined 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#030517",
          "borderRadius": 10,
          "height": 8,
          "width": 20,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#8F8F8F",
          "borderRadius": 10,
          "height": 8,
          "marginLeft": 2,
          "width": 8,
        },
      ]
    }
  />
</View>
`;
